### PR TITLE
Keep progress in notification scroll when updating

### PR DESCRIPTION
**Pull request by yairm210**
Mentioned in Discord

### CODE REVIEW

File: core/src/com/unciv/ui/worldscreen/NotificationsScroll.kt [hunk 1]
```diff
@@ -102,7 +107,7 @@ class NotificationsScroll(
         pack()
         if (height == newHeight) return
         height = newHeight  // after this, maxY is still incorrect until layout()
-        invalidateHierarchy()
+        layout()
```
[the remark this answers is not in the document]
Comment (yairm210) [reply core/src/com/unciv/ui/worldscreen/NotificationsScroll.kt:110]: What we need is the layout, so we can just call it directly, there's no need to do it in a roundabout way
Practically they are the same though

[the remark this answers is not in the document]
Comment (yairm210) [reply core/src/com/unciv/ui/worldscreen/NotificationsScroll.kt:110]: Fair enough

**Comment by Ouaz** [comment]
Just a question: what is this supposed to do exactly?

In 4.20, when I scroll the notification list, and I click a notification, the notification list scroll progress is reset, and doesn't keep the position where I scrolled it. So I still have to scroll the list again, to be able to click the next notification on the list.

I think this commit would have fixed that given its name "keep notification scroll progress" ^^

**Comment by yairm210** [comment]
That's what is does do for me
Can you send me your game save? Did this happen with all notifications or only some?

**Pull request merged**